### PR TITLE
Amazon Athena data source

### PR DESCRIPTION
**Pull request by gruuya**
- Add another S3-focused data source, this time backed by Amazon Athena, internally employing our SQLAlchemy FDW
- Install missing libs in the engine Dockerfile (PyAthena, pandas)
- Fix CSV mount example, and make Snowflake plugin default to batch size of 10000 if none specified

### CODE REVIEW

File: splitgraph/ingestion/athena/__init__.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+}
+EOF
+```
+    """
+
+    commandline_kwargs_help: str = (
+        build_commandline_help(credentials_schema) + "\n" + build_commandline_help(params_schema)
+    )
+
+    _icon_file = "snowflake.svg"
```
Comment (mildbyte) [inline splitgraph/ingestion/athena/__init__.py:72]: Use something like https://symbols.getvecta.com/stencil_5/0_aws-athena.bb0d0ced14.svg?

File: splitgraph/ingestion/csv/__init__.py [hunk 2]
```diff
@@ -241,23 +241,29 @@ class CSVDataSource(ForeignDataWrapperDataSource):
...
-    "autodetect_header": true,
-    "autodetect_dialect": true,
-    "autodetect_encoding": true
-  }
+{
+	"s3_access_key": "ABCD",
+	"s3_secret_key": "abcd",
+	"connection":
+	{
+		 "connection_type": "s3",
```
Comment (mildbyte) [inline splitgraph/ingestion/csv/__init__.py:256]: indentation

File: splitgraph/ingestion/athena/__init__.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+
+class AmazonAthenaDataSource(ForeignDataWrapperDataSource):
+    credentials_schema: Dict[str, Any] = {
+        "type": "object",
+        "properties": {
+            "aws_access_key_id": {"type": "string", "title": "AWS Access Key Id"},
+            "aws_secret_access_key": {"type": "string", "title": "AWS Secret Access Key"},
+        },
+    }
```
Comment (mildbyte) [inline splitgraph/ingestion/athena/__init__.py:22]: These should be required, e.g.:

```suggestion
        "required": ["aws_access_key_id", "aws_secret_access_key"],
    }
```

File: splitgraph/ingestion/athena/__init__.py [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+        if "schema" in self.params:
+            options["schema"] = self.params["schema"]
+
+        return options
+
+    def _build_db_url(self) -> str:
+        """Construct the SQLAlchemy Amazon Athena db_url"""
+
+        aws_access_key_id = self.params["aws_access_key_id"]
+        aws_secret_access_key = self.params["aws_secret_access_key"]
```
Comment (mildbyte) [inline splitgraph/ingestion/athena/__init__.py:120]: I think this comes from credentials (not merged into params)

```suggestion
        aws_access_key_id = self.credentials["aws_access_key_id"]
        aws_secret_access_key = self.credentials["aws_secret_access_key"]
```

**Review by mildbyte — APPROVED** [review]

File: splitgraph/ingestion/athena/__init__.py (hunk 4 above)
Comment (gruuya) [reply splitgraph/ingestion/athena/__init__.py:120]: Curiously, it actually works properly either way.

**Pull request merged**